examples/DivisorCurve.jl

```diff
@@ -0,0 +1,329 @@
+################################################################################
+
+function curve(D::CurveDivisor)
+    return D.C
+end
+
+function Oscar.degree(D::CurveDivisor)
+    return D.degree
+end
+
+################################################################################
+
+function _check_same_curve(D::CurveDivisor, E::CurveDivisor)
+    return D.C == E.C
+end
+
+################################################################################
+################################################################################
+# Sum of two divisors
+
+function Base.:+(D::AffineCurveDivisor, E::AffineCurveDivisor)
+    _check_same_curve(D, E) || error("The divisors are not on the same curve")
+    return AffineCurveDivisor(D.C, mergewith(+, D.divisor, E.divisor))
+end
+
+function Base.:+(D::ProjCurveDivisor, E::ProjCurveDivisor)
+    _check_same_curve(D, E) || error("The divisors are not on the same curve")
+    return ProjCurveDivisor(D.C, mergewith(+, D.divisor, E.divisor))
+end
+
+################################################################################
+
+function Base.:-(D::CurveDivisor, E::CurveDivisor)
+    return D + -1*E
+end
+
+################################################################################
+################################################################################
+
+function Base.:*(k::Int, D::AffineCurveDivisor{S}) where S <: FieldElem
+    if k == 0
+        return AffineCurveDivisor(D.C, Dict{Point{S}, Int}())
+    elseif k == 1
+        return D
+    else
+        return AffineCurveDivisor(D.C, Dict((P, k*m) for (P, m) in D.divisor))
+    end
+end
+
+function Base.:*(k::Int, D::ProjCurveDivisor{S}) where S <: FieldElem
+    if k == 0
+        return ProjCurveDivisor(D.C, Dict{Oscar.Geometry.ProjSpcElem{S}, Int}())
+    elseif k == 1
+        return D
+    else
+        return ProjCurveDivisor(D.C, Dict((P, k*m) for (P, m) in D.divisor))
+    end
+end
+
+################################################################################
+
+function ==(D::CurveDivisor, E::CurveDivisor)
+    return D.C == E.C && D.divisor == E.divisor
+end
+
+################################################################################
+################################################################################
+
+@doc Markdown.doc"""
+    iseffective(D::CurveDivisor)
+
+Return `true` if `D` is an effective divisor, `false` otherwise.
+"""
+function iseffective(D::CurveDivisor)
+    return all(v >= 0 for v in values(D.divisor))
+end
+
+################################################################################
+################################################################################
+
+@doc Markdown.doc"""
+    multiplicity(C::AffinePlaneCurve, phi::AbstractAlgebra.Generic.Frac, P::Point)
+
+Return the multiplicity of the rational function `phi` on the curve `C` at the point `P`.
+"""
+function multiplicity(C::AffinePlaneCurve, phi::AbstractAlgebra.Generic.Frac, P::Point)
```

Comment:
Optional comment:

Shouldn't this ensure the types of C and S match, like the next method? Something like this:
```suggestion
function multiplicity(C::AffinePlaneCurve{S}, phi::AbstractAlgebra.Generic.Frac, P::Point{S}) where S <: FieldElem
```
(dunno if there is a corresponding type parameter for `Frac`?)

Reply:
Do we have function fields of curves already? Then phi would be in that function field and C would be a redundant argument?

Reply:
I have a problem with taking elements in a function fields, that's why I took fractions. I just wrote an  #241 for my problem.